frameworklauncher/src/main/java/com/microsoft/frameworklauncher/service/Service.java

```diff
@@ -251,20 +251,17 @@ private ContainerLaunchContext setupContainerLaunchContextForDefaultAM(
     // SetupLocalEnvironment
     Map<String, String> localEnvs = new HashMap<>();
 
-    StringBuilder classpath = new StringBuilder(ApplicationConstants.Environment.CLASSPATH.$$())
-        .append(ApplicationConstants.CLASS_PATH_SEPARATOR).append("./*");
+    // Internal class is preferred over external class
+    localEnvs.put(ApplicationConstants.Environment.CLASSPATH_PREPEND_DISTCACHE.name(), "true");
```

Comment:
>localEnvs [](start = 4, length = 9)

Do you think we can create a unit test to make sure the Windows/Linux OS env. won't break this?

Reply:
Good point. But it is hard to UT in Launcher, since it at least need to mock a NM, and do the same UT as NM did. Since it is a feature in NM, we can just believe the NM UT is passed.

---
In reply to: [159354155](https://github.com/Microsoft/pai/pull/156#discussion_r159354155) [](ancestors = 159354155)

Reply:
let's create an issue to track this then.

---
In reply to: [159355911](https://github.com/Microsoft/pai/pull/156#discussion_r159355911) [](ancestors = 159355911,159354155)